tests/indexes_/test_atlas_indexes.py

```diff
@@ -0,0 +1,218 @@
+                fields=["headline", "number", "body", "data", "embedded", "auto_now"],
+            )
+            editor.add_index(index=index, model=Article)
+            index_info = connection.introspection.get_constraints(
+                cursor=None,
+                table_name=Article._meta.db_table,
+            )
+            expected_options = {
+                "dynamic": False,
+                "fields": {
+                    "auto_now": {"type": "date"},
+                    "body": {
+                        "indexOptions": "offsets",
+                        "norms": "include",
+                        "store": True,
+                        "type": "string",
+                    },
+                    "data": {"dynamic": False, "fields": {}, "type": "document"},
+                    "embedded": {"dynamic": False, "fields": {}, "type": "embeddedDocuments"},
+                    "headline": {
+                        "indexOptions": "offsets",
+                        "norms": "include",
+                        "store": True,
+                        "type": "string",
+                    },
+                    "number": {
+                        "indexDoubles": True,
+                        "indexIntegers": True,
+                        "representation": "double",
+                        "type": "number",
+                    },
+                },
+            }
+            self.assertCountEqual(index_info[index.name]["columns"], index.fields)
+            self.assertEqual(index_info[index.name]["options"], expected_options)
+            self.assertAddRemoveIndex(editor, Article, index)
+
+    def test_field_not_exists(self):
+        index = AtlasSearchIndex(
+            name="recent_article_idx",
+            fields=["headline", "number1"],
+        )
+        with connection.schema_editor() as editor:
+            msg = "Article has no field named 'number1'"
+            with self.assertRaisesMessage(
+                FieldDoesNotExist, msg
+            ), connection.schema_editor() as editor:
+                editor.add_index(index=index, model=Article)
+
+
+class AtlasIndexTestsWithData(AtlasIndexTests):
+    @classmethod
+    def setUpTestData(cls):
+        articles = [
+            Article(
+                headline=f"Title {i}",
+                number=i,
+                body=f"body {i}",
+                data="{json: i}",
+                embedded=Data(integer=i),
+                auto_now=datetime.datetime.now(),
+                title_embedded=[0.1] * 10,
+                description_embedded=[2.5] * 10,
+                number_list=[2] * i,
+                name_list=[f"name_{i}"] * 10,
+            )
+            for i in range(5)
+        ]
+        cls.objs = Article.objects.bulk_create(articles)
+
+
+class AtlasSearchIndexTests(TestCase):
+    # Schema editor is used to  create the index to test that it works.
+    # available_apps = ["indexes"]
+    available_apps = None  # could be removed?
```

Comment:
It's not currently that helpful, but I'd include it for future possibilities. Without it, if we try to run Django's test suite without arguments, Django has to flush all test apps between each test (see  86d5885b34855a123756462f9d6954eb9420f763). If we added `available_apps` to all of Django's `TestCase` classes, then it would be feasible to run the Django test suite (without any test app names).